### PR TITLE
Confirmation message for successful section assignment on Curriculum Catalog page

### PR DESCRIPTION
**Pull request by TurnerRiley**
Displays a confirmation message for when a user successfully assigns a course to one of their sections:
Success_message

## Shows message on successful assignment
https://github.com/code-dot-org/code-dot-org/assets/56283563/6f19f12b-7462-453e-8e27-05960752802e

## Does not show message if no assignment changes were made
https://github.com/code-dot-org/code-dot-org/assets/56283563/783df697-27c8-4036-85f6-bb473e9ceb17

## Does not show message on failed assignment
For this one, I was having trouble finding a way to simulate a failed assignment so I just had the assignment process throw an arbitrary error.
https://github.com/code-dot-org/code-dot-org/assets/56283563/4c2e3be7-4cca-47fb-85a4-a10a750925e8

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-643&assignee=60d62161f65054006980bd71)

## Testing story
Local testing and creating a [Jira ticket to add unit tests](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-704&assignee=60d62161f65054006980bd71). We are separating out the test-writing to ensure this change makes it in with tomorrow's deploy with all the other parts of the curriculum launch.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx [hunk 1]
```diff
@@ -309,6 +322,20 @@ const CurriculumCatalog = ({
...
         imageUrl={CourseCatalogIllustration01}
       />
+      {showAssignSuccessMessage && (
+        <div className={style.assignSuccessMessageContainer}>
+          <BodyOneText className={style.assignSuccessMessage}>
+            {assignSuccessMessage}
+          </BodyOneText>
+          <button
+            aria-label="close success message"
+            onClick={() => setShowAssignSuccessMessage(false)}
```
Comment (megcrenshaw) [inline apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx:332]: Should this also do setAssignSuccessMessage('') to reset that as well? (I don't think this is urgent / worthy of a drone re-run)

File: apps/src/templates/MultipleSectionsAssigner.jsx [hunk 2]
```diff
@@ -154,6 +155,30 @@ const MultipleSectionsAssigner = ({
...
+    scriptId
+  ) => {
+    onAssignSuccess
+      ? assignToSection(
+          sectionId,
+          courseId,
+          courseOfferingId,
+          courseVersionId,
+          scriptId
+        ).then(onAssignSuccess)
```
Comment (megcrenshaw) [inline apps/src/templates/MultipleSectionsAssigner.jsx:172]: I never thought of assignToSection as a promise, but this makes sense! Cool beans 😎

**Review by megcrenshaw — APPROVED** [review]
Clever approach! Looks great, and thanks for adding the ticket for test coverage

File: apps/src/templates/MultipleSectionsAssigner.jsx (hunk 2 above)
Comment (TurnerRiley) [reply apps/src/templates/MultipleSectionsAssigner.jsx:172]: I didn't either but I believe that's the way it ends up being handled!

File: apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx:332]: Good call! I'll add that as a note in the test ticket as part of the cleanup!

**Pull request merged**